packages/rrweb/src/record/observer.ts

```diff
@@ -54,10 +54,36 @@ type WindowWithAngularZone = IWindow & {
 export const mutationBuffers: MutationBuffer[] = [];
 export const processedNodeManager = new ProcessedNodeManager();
 
-const isCSSGroupingRuleSupported = typeof CSSGroupingRule !== 'undefined';
-const isCSSMediaRuleSupported = typeof CSSMediaRule !== 'undefined';
-const isCSSSupportsRuleSupported = typeof CSSSupportsRule !== 'undefined';
-const isCSSConditionRuleSupported = typeof CSSConditionRule !== 'undefined';
+function isCSSStyleSheetMonkeyPatchable(
+  prop:
+    | 'CSSGroupingRule'
+    | 'CSSMediaRule'
+    | 'CSSSupportsRule'
+    | 'CSSConditionRule',
+): boolean {
+  /* eslint-disable @typescript-eslint/ban-ts-comment,@typescript-eslint/unbound-method */
+  return Boolean(
+    typeof window[prop] !== 'undefined' &&
+      // Note: Generally, this check _shouldn't_ be necessary
+      // However, in some scenarios (e.g. jsdom) this can sometimes fail, so we check for it here
+      window[prop].prototype &&
+      // @ts-ignore ensure it is actually set
+      window[prop].prototype.insertRule &&
+      window[prop].prototype.deleteRule,
+  );
+  /* eslint-enable @typescript-eslint/ban-ts-comment,@typescript-eslint/unbound-method */
```

Comment:
`ts-ignore` is a little aggressive, this would be a much cleaner way of getting the same result:
```suggestion
  return Boolean(
    typeof window[prop] !== 'undefined' &&
      // Note: Generally, this check _shouldn't_ be necessary
      // However, in some scenarios (e.g. jsdom) this can sometimes fail, so we check for it here
      window[prop].prototype &&
      "insertRule" in window[prop].prototype &&
      "deleteRule" in window[prop].prototype,
  );
```